src/auth/mod.rs

```diff
@@ -45,6 +45,8 @@ impl Token {
 
 #[derive(Clone, Debug)]
 pub struct GoogleCloudAuth {
+    // XXX(don) we only cache a token for the last set of scopes requested.
+    scopes: Arc<RwLock<Vec<String>>>,
```

Comment:
Why not use an `Arc<RwLock<(..., ...)>>`. These are always read and written together.